examples/synapseml/autologging.md

```diff
@@ -0,0 +1,89 @@
+## MlFlow automatic Logging with SynapseML
+
+[MLflow automatic logging](https://www.mlflow.org/docs/latest/tracking.html#automatic-logging) allows you to log metrics, parameters, and models without the need for explicit log statements.
+SynapseML supports autologging for every model in the library.
+
+Install SynapseML library following this [guidance](https://microsoft.github.io/SynapseML/docs/getting_started/installation/)
+
+To enable autologging for SynapseML:
+
+1. Download this customized [log_model_allowlist file](https://mmlspark.blob.core.windows.net/publicwasb/log_model_allowlist.txt) and put it at a place that your code has access to.
+   For example:
+
+- In Synapse `wasb://<containername>@<accountname>.blob.core.windows.net/PATH_TO_YOUR/log_model_allowlist.txt`
+- In Databricks `/dbfs/FileStore/PATH_TO_YOUR/log_model_allowlist.txt`.
```

Comment:
@serena-ruan would you like to include synapse.ml.* in https://github.com/mlflow/mlflow/blob/master/mlflow/pyspark/ml/log_model_allowlist.txt?

Reply:
Yes definitely! But we do have some models that're as big as ALS models, we could exclude them outside of the default log_model_allowlist.txt file and if users are not against autologging them they can replace with their own log_model_allowlist file.

Reply:
That sounds great - let's include the smaller ones in the allowlist by default. Also, do you think that introducing a `log_model_allowlist` argument to `mlflow.pyspark.autolog()` that accepts a Python list would make it easier for users to specify which models they want to log?

Reply:
Yes I agree! Actually maybe an argument like 'extra_log_model_allowlist' would be helpful if users just want to add some other models. If they have a persistent model list they want to log, we already support reading from url and they just need to change a spark config which is way simpler.

Reply:
Awesome! I think calling it `log_model_allowlist` is okay; it can override the file. Can you introduce that argument to this PR and also add small Synapse models to https://github.com/mlflow/mlflow/blob/master/mlflow/pyspark/ml/log_model_allowlist.txt?